### PR TITLE
Dockerfile follow-up fixes and improvements

### PR DESCRIPTION
**Pull request by thaJeztah**
Whoops; noticed that my previous change in https://github.com/docker/docker.github.io/pull/8413 introduced a small bug; fixing it in this PR; the fix is in the first commit; other changes are follow-ups to make some small improvements (to be used in https://github.com/docker/docker.github.io/pull/8414)

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @L-Hudson @bermudezmt ptal

File: _scripts/fetch-library-samples.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+# Get the Library docs
+svn co https://github.com/docker-library/docs/trunk ./_samples/library || (echo "Failed library download" && exit 1)
```
Comment (bermudezmt) [inline _scripts/fetch-library-samples.sh]: @thaJeztah sorry this hasn't been communicated clearly in a previous PR. We have updated the library logic to have the sample library links point to Hub pages instead. I don't think fetching them is still necessary except for archived versions. Let me double check on this though.

Comment (thaJeztah) [reply _scripts/fetch-library-samples.sh]: Yeah, I never liked the samples being there, but it was still in the dockerfile, so I kept it

Comment (thaJeztah) [reply _scripts/fetch-library-samples.sh]: oh, looks like the "samples home" https://docs.docker.com/samples/ links to the docker hub pages, but the navigation still has the "fetched" versions as well

screenshot 2019-03-07 at 23 26 35

And those pages are still active; https://docs.docker.com/samples/library/alpine/

I can work on turning those into redirects, and remove them from the actual docs (that may need a bit of thinking, so I'd prefer doing that in a follow-up to not uphold the fixes in this PR)

Comment (bermudezmt) [reply _scripts/fetch-library-samples.sh]: Both of those comments are accurate - just haven't had the time to come back and clean those up. :( Thank you - will merge this one then!

**Pull request merged**